ffmpeg_utils.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+from datetime import datetime, timedelta
+import subprocess
+
+from constants import RAW_VIDEOS_FRAMERATE
+
+
+def subprocess_run_with_guardfile(cmd: list[str], guardfile_path: Path):
+    """
+    Runs a subprocess command, and writes the command to a guardfile, which is deleted after subprocess runs
+    successfully as a way to monitor whether there was a crash mid-processing, or a forced stop.
+
+    """
+    running_message: str = "Running: " + " ".join(cmd)
+    with open(guardfile_path, "w") as f:
+        f.write(running_message)
+    subprocess.run(cmd, check=True)
+    guardfile_path.unlink()
+
+
+def get_video_duration_in_seconds(file_path: Path) -> float:
+    cmd = [
+        "ffprobe",
+        "-v",
+        "error",
+        "-show_entries",
+        "format=duration",
+        "-of",
+        "default=noprint_wrappers=1:nokey=1",
+        file_path,
+    ]
+    output = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    return float(output.stdout)
+
+
+def extract_timecode_from_conflab_raw_go_pro_video_metadata(
+    video_path: Path,
+) -> datetime:
+    cmd = [
+        "ffprobe",
+        "-hide_banner",
+        "-show_streams",
+        "-i",
+        str(video_path),
+    ]
+    # fmt: on
+    res = subprocess.run(cmd, capture_output=True)
+    video_timecode = res.stdout.decode("utf-8").rstrip()
+    # Find TAG:timecode=<VAL> in the string
+    video_timecode = video_timecode.split("TAG:timecode=")[1].split("\n")[0]
+    video_date = [2019, 10, 24]
```

Comment:
Should we add two comments here, one saying the that timecode in the mp4 doesn't have the date but only the HH:MM:SS and another saying the the 2 hours is to account for the timezone